lib/api/api_auth.go

```diff
@@ -37,15 +37,72 @@ func emitLoginAttempt(success bool, username, address string, evLogger events.Lo
 	}
 }
 
+func antiBruteForceSleep() {
+	time.Sleep(time.Duration(rand.Intn(100)+100) * time.Millisecond)
+}
+
+func unauthorized(w http.ResponseWriter) {
+	antiBruteForceSleep()
+	w.Header().Set("WWW-Authenticate", "Basic realm=\"Authorization Required\"")
+	http.Error(w, "Not Authorized", http.StatusUnauthorized)
+}
+
+func forbidden(w http.ResponseWriter) {
+	antiBruteForceSleep()
+	http.Error(w, "Forbidden", http.StatusForbidden)
+}
+
+func equalsAny(s string, values []string) bool {
+	for _, value := range values {
+		if s == value {
+			return true
+		}
+	}
+	return false
+}
```

Comment:
This can be `slices.Contains`

Reply:
Thanks! Fixed in d03dda5fa10be625f40a7d688b03dd48df40091b.

Reply:
Sorry, I'm not familiar enough with Go - how should I fix this? https://github.com/syncthing/syncthing/actions/runs/6430309806/job/17461052511?pr=8757

Do I add something to `go.mod` or the like, or should I use `"golang.org/x/exp/slices"` instead (as in [`api_test.go`](https://github.com/syncthing/syncthing/blob/4f6b86a1c09b99aeb239d49c92ea7036f7a24e9e/lib/api/api_test.go#L48))?

Reply:
Yes it needs to be spelled `golang.org/x/exp/slices` for now

Reply:
Ok, fixed in 6e0b60b73fd53ea46417a24125be194d8a6b2af3.